tests/nutter/test_testresult.py

```diff
@@ -139,3 +160,15 @@ def test__deserialize__data_is_base64_str__can_deserialize():
     test_results_from_data = TestResults().deserialize(serialized_str)
 
     assert test_results == test_results_from_data
+
+
+def get_py4j_exception():
+    # Raise a Py4JJavaError
+    # NB: a running Py4J Java Gateway is required for this to work.
+    gateway = JavaGateway()
+    random = gateway.jvm.java.util.Random()
+    try:
+        _ = random.nextInt(-1)
+    except Py4JJavaError as e:
+        _ex = e
+    return _ex

```

Comment:
Would it be possible to just throw a `Py4JJavaError` something like this: `raise Py4JJavaError()` ?

Reply:
It's unfortunately a bit more complicated, but I found a way to create a `Py4JJavaError` by mocking a bunch of other Py4J objects... I pushed the changes and removed the dependency to a real Py4J environment.

Link to SO post for reference: https://stackoverflow.com/questions/58384719/pyspark-mocking-exception-test-succeeds-but-exception-is-not-handled